x-pack/plugins/security/public/components/management/users/users.js

```diff
@@ -0,0 +1,225 @@
+        this.setState({ permissionDenied: true });
+        setTimeout(() => {
+          changeUrl('/management');
+        }, 3000);
+      } else {
+        toastNotifications.addDanger(`Error fetching users: ${e.data.message}`);
+      }
+    }
+  }
+  renderToolsLeft() {
+    const { selection } = this.state;
+    if (selection.length === 0) {
+      return;
+    }
+    const numSelected = selection.length;
+    return (
+      <EuiButton
+        data-test-subj="deleteUserButton"
+        color="danger"
+        onClick={() => this.setState({ showDeleteConfirmation: true })}
+      >
+        Delete {numSelected} user{numSelected > 1 ? 's' : ''}
+      </EuiButton>
+    );
+  }
+  onCancelDelete = () => {
+    this.setState({ showDeleteConfirmation: false });
+  }
+  render() {
+    const { users, permissionDenied, showDeleteConfirmation, selection } = this.state;
+    const { apiClient } = this.props;
+    if (permissionDenied) {
+      return (
+        <EuiPage className="mgtUsersListingPage">
+          <EuiPageBody>
+            <EuiPageContent verticalPosition="center" horizontalPosition="center" className="mgtUsersListingPage__content">
+              <EuiEmptyPrompt
+                iconType="securityApp"
+                iconColor={null}
+                title={<h2>Permission denied</h2>}
+                body={<p data-test-subj="permissionDeniedMessage">You do not have permission to manage users.</p>}
+              />
+            </EuiPageContent>
+          </EuiPageBody>
+        </EuiPage>
+      );
+    }
+    const path = '#/management/security/';
+    const columns = [
+      {
+        field: 'full_name',
+        name: 'Full Name',
+        sortable: true,
+        truncateText: true,
+        render: fullName => {
+          return <div data-test-subj="userRowFullName">{fullName}</div>;
+        },
+      },
+      {
+        field: 'username',
+        name: 'User Name',
+        sortable: true,
+        truncateText: true,
+        render: username => (
+          <EuiLink data-test-subj="userRowUserName" href={`${path}users/edit/${username}`}>
+            {username}
+          </EuiLink>
+        ),
+      },
+      {
+        field: 'roles',
+        name: 'Roles',
+        render: rolenames => {
+          const roleLinks = rolenames.map((rolename, index) => {
+            return (
+              <Fragment key={rolename}>
+                <EuiLink href={`${path}roles/edit/${rolename}`}>{rolename}</EuiLink>
+                {index === rolenames.length - 1 ? null : ', '}
+              </Fragment>
+            );
+          });
+          return <div data-test-subj="userRowRoles">{roleLinks}</div>;
+        },
+      },
+      {
+        field: 'metadata._reserved',
+        name: 'Reserved',
+        sortable: false,
+        width: '100px',
+        align: 'right',
+        render: reserved =>
+          reserved ? <EuiIcon data-test-subj="reservedUser" type="lock" /> : null,
+      },
+    ];
```

Comment:
Makes sense, but the problem is that the actions should not include delete for built in users, and there's not really a way to vary the menu based on the item in EUI at the moment.  There's a render for each menu item, but returning null from that leaves a blank menu item:
![image](https://user-images.githubusercontent.com/2479496/42887448-fc678506-8a73-11e8-90ba-fe9b2847b6bb.png)

Reply:
Jumping in here out of context, but you CAN disable the actions if that helps.

Reply:
@snide I need to only disable specific actions, not all of them.  And I want them not to show in the UI at all.

Reply:
@bmcconaghy It can be done using either `available` or `enabled` prop on a `DefaultItemAction` like this:
```js
const actions = {
    actions: [
        {
            name: 'Edit',
            ...
        },
        {
            name: 'Delete',
            available: (user) => !user.metadata._reserved
            ...
        },
    ]
}
```

Reply:
Didn't know that, thanks for the knowledge @jen-huang .  I'm still sticking with the current non actions behavior, as it seems silly to have an action dropdown for 2 (and in the case of reserved users, 1) action.